autoPyTorch/data/tabular_feature_validator.py

```diff
@@ -323,14 +348,17 @@ def _get_columns_info(
 
         # Make sure each column is a valid type
         for i, column in enumerate(X.columns):
```

Comment:
Could you change this for loop block to the following?
The if statement should be coded in this way and I made the error message shorter by sharing a common part.
```suggestion
        for i, column in enumerate(X.columns):
            if self.all_nan_columns is not None and column in self.all_nan_columns:
                continue

            column_dtype = self.dtypes[i]
            err_msg = "Valid types are `numerical`, `categorical` or `boolean`, " \
                      "but input Column {} has an invalid type `{}`.".format(column, column_dtype)

            if column_dtype in ['category', 'bool']:
                categorical_columns.append(column)
                feat_type.append('categorical')
            # Move away from np.issubdtype as it causes
            # TypeError: data type not understood in certain pandas types
            elif is_numeric_dtype(column_dtype):
                feat_type.append('numerical')
                numerical_columns.append(column)
            elif column_dtype == 'object':
                # TODO verify how would this happen when we always convert the object dtypes to category
                raise ValueError(
                    "{} Cast it to a valid dtype before feeding it to AutoPyTorch. "
                    "You can cast it to a valid dtype using pandas.Series.astype."
                    "If you are working with string objects, the following "
                    "tutorial illustrates how to work with text data: "
                    "https://scikit-learn.org/stable/tutorial/text_analytics/working_with_text_data.html".format(
                        # noqa: E501
                        err_msg,
                    )
                )
            elif pd.core.dtypes.common.is_datetime_or_timedelta_dtype(column_dtype):
                raise ValueError(
                    "{} Convert the time information to a numerical value"
                    " before feeding it to AutoPyTorch. "
                    "One example of the conversion can be found on "
                    "https://stats.stackexchange.com/questions/311494/".format(err_msg)
                )
            else:
                raise ValueError(
                    "{} Make sure your data is formatted in a correct way"
                    "before feeding it to AutoPyTorch.".format(err_msg)
                )
```

Reply:
I have done so